### ISSUE TITLE
UWP: System.Net.Http.HttpClient does not clear Authorization Header for redirect

### ISSUE THREAD

**Issue by jNery** [dotnet/corefx#32175]
This is not reproducible in `Windows.Web.Http.HttpClient `  or in `WPF: System.NET.Http.HttpClient`. The forwarded request have no Authorization header set.

Only in `UWP: System.Net.Http.HttpClient`

To reproduce:
1 - Create a handler that will set Authorization header based on request
2 - Use url that requires Authorization header, request should succeed
3 - Use url that requires Authorization header but forwards to a different host (i.e. REST API call that goes to Amazon Web Services), request fails because header was re-used with a different host.

```CSharp
var handler = new MyHttpClientHandler();
var client = new System.Net.Http.HttpClient(handler);

// Request with Token that will not cause re-direct succeeds
var json = await client.GetAsync(new Uri($"{itemUrl}?token={oAuthCred.Token}&f=json"));
var content = await json.Content.ReadAsStringAsync();
System.Diagnostics.Debug.WriteLine(content);

// Request with Token that will cause re-direct fails with 400 BadRequest
var data = await client.GetAsync(new Uri($"{itemUrl}/data?token={oAuthCred.Token}"));
System.Diagnostics.Debug.WriteLine($"Status code : {data.StatusCode}");
```

HttpClientHandler sets Authorization header as required by item url
```CSharp
public class MyHttpClientHandler : HttpClientHandler
{
    public MyHttpClientHandler() { }
    protected override Task<System.Net.Http.HttpResponseMessage> SendAsync(System.Net.Http.HttpRequestMessage request, CancellationToken cancellationToken)
    {
        var parts = request.RequestUri.OriginalString.Split("token=");
        if (parts.Length > 1)
        {
            var token = parts[1];
            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", token);
        }
        return base.SendAsync(request, cancellationToken);
    }
}
```

**Comment by davidsh** [comment]
cc: @karelz @rmkerr

**Comment by karelz** [comment]
We are aware of this issue, the fix is on its way and should be available next month.
Thanks for your report!

**Comment by karelz** [comment]
Fixed in UWP 6.1.9: https://github.com/Microsoft/dotnet/blob/master/releases/UWP/net-native2.1/README.md
See #32730 for details.

**Comment by chrisahill1** [comment]
@karelz looks like this fix was regressed in 6.2.2 after upgrading from 6.1.9 I'm seeing my bearer token added to my request in UWP application to amazon instance after being redirected. My original request is a 302 Not found with a redirect link to amazon instance the bearer token for the original request is being added to the redirect link to amazon but there are already credentials for the amazon uri in the bearer token is not suppose to be there. the bearer token was an OAuth token for the original request issued, but when redirect to the amazon instance the bearer token was being added. Please confirm you didn't regress this bug fix in 6.2.2
